src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
```

Comment:
> all I “know” is that some jobs are “normal” (= decent) and some others are "big" (will fail with less than, say, 4 GB allocated to them and their size increases with data size).

Why not write those, or better yet, if you don't know _don't_ add unneeded complexity until we need it. As a user, unless you're sitting next to me or on slack, how do I know whether `rsync` is `Decent` or `Big`. Why add an extra option just so that a user can make a mistake?